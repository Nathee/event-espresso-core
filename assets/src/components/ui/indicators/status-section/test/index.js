/**
 * External imports
 */
import { shallow } from 'enzyme';

/**
 * Internal imports
 */
import StatusSection from '../index';

describe( 'StatusSection', () => {
	const getWrapper = ( props ) => shallow( <StatusSection
		statusLabel={ 'Transaction Status:' }
		statusValue={ 'Failed' }
		statusCode={ 'TFL' }
		{ ...props }
	/> );
	it( 'returns expected shape for default props (with required ' +
		'values)', () => {
		expect( getWrapper() ).toMatchSnapshot();
	} );
	it( 'returns expected classname for section container when custom css ' +
		'class passed in', () => {
		const wrapper = getWrapper( { className: 'custom' } );
		expect(
			wrapper.find( '.ee-status-section-container' ).hasClass( 'custom' )
		).toBe( true );
	} );
} );