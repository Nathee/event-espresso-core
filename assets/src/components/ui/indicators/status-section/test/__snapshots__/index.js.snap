// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusSection returns expected shape for default props (with required values) 1`] = `
<section
  className="ee-status-section-container"
>
  <span
    className="ee-status-section-label"
  >
    Transaction Status:
  </span>
  <span
    className="ee-status-section-value"
  >
    Failed
  </span>
  <StatusCircle
    circleSize="circle-size-20"
    className=""
    statusCode="TFL"
    statusLabel="Transaction Status:"
  />
</section>
`;
